src/rendering/renderers/shared/texture/RenderableGCSystem.ts

```diff
@@ -0,0 +1,192 @@
+    private _now: number;
+
+    /** @param renderer - The renderer this System works for. */
+    constructor(renderer: Renderer)
+    {
+        this._renderer = renderer;
+    }
+
+    public init(options: RenderableGCSystemOptions): void
+    {
+        options = { ...RenderableGCSystem.defaultOptions, ...options };
+
+        this.maxUnusedTime = options.renderablesGCMaxUnusedTime;
+        this._frequency = options.renderableGCFrequency;
+
+        this.enabled = options.renderablesGCActive;
+    }
+
+    get enabled(): boolean
+    {
+        return !!this._handler;
+    }
+
+    set enabled(value: boolean)
+    {
+        if (this.enabled === value) return;
+
+        if (value)
+        {
+            this._handler = this._renderer.scheduler.repeat(
+                () => this.run(),
+                this._frequency
+            );
+        }
+        else
+        {
+            this._renderer.scheduler.cancel(this._handler);
+        }
+    }
+
+    public prerender(): void
+    {
+        this._now = performance.now();
+    }
+
+    public addRenderable(renderable: Renderable, instructionSet: InstructionSet): void
+    {
+        renderable._lastUsed = this._now;
+
+        if (renderable._lastInstructionTick === -1)
+        {
+            // TODO manage index...
+            this._managedRenderables.push(renderable);
+        }
+
+        renderable._lastInstructionTick = instructionSet.tick;
+    }
+
+    /** Runs the scheduled garbage collection */
+    public run(): void
+    {
+        const now = performance.now();
+
+        const managedRenderables = this._managedRenderables;
+
+        const renderPipes = this._renderer.renderPipes;
+
+        let offset = 0;
+
+        for (let i = 0; i < managedRenderables.length; i++)
+        {
+            const renderable = managedRenderables[i];
+
+            const renderGroup = renderable.renderGroup ?? renderable.parentRenderGroup;
+            const currentIndex = renderGroup?.instructionSet?.tick ?? -1;
+
+            if (renderable._lastInstructionTick !== currentIndex && now - renderable._lastUsed > this.maxUnusedTime)
+            {
+                if (!renderable.destroyed)
+                {
+                    const rp = renderPipes as unknown as Record<string, RenderPipe>;
+
+                    rp[renderable.renderPipeId].destroyRenderable(renderable);
+                }
+
+                // remove from the array as this has been destroyed..
+                renderable._lastInstructionTick = -1;
+                offset++;
+            }
+            else
+            {
+                managedRenderables[i - (offset)] = renderable;
+            }
+        }
+
+        managedRenderables.length = managedRenderables.length - offset;
+    }
+
+    public destroy(): void
+    {
+        this.enabled = false;
+        this._renderer = null as any as Renderer;
+        this._managedRenderables.length = 0;
+    }
+}
+
+extensions.add(RenderableGCSystem);
```

Comment:
this should be added to `SharedSystems`?

Reply:
i was doing both! fixed now :)